azkeys/coseSigner.go

```diff
@@ -20,32 +20,83 @@ import (
 	"github.com/veraison/go-cose"
 )
 
-// CoseSignerKeyVault is the azure keyvault client for interacting with keyvault keys
-// using a cose.Signer interface
-type CoseSignerKeyVault struct {
-	keyName string
-	alg     keyvault.JSONWebKeySignatureAlgorithm
-	*KeyVault
```

Comment:
This is all very clear and nice. Great commentary identifying the purpose of the types